driver-kotlin-sync/src/main/kotlin/com/mongodb/kotlin/client/MongoCursor.kt

```diff
@@ -61,13 +55,81 @@ public open class MongoCursor<T : Any>(private val wrapped: JMongoCursor<T>) : I
      * @return the next element in the iteration if available or null.
      * @see [Tailable Cursor](https://www.mongodb.com/docs/manual/reference/glossary/#term-tailable-cursor)
      */
-    public fun tryNext(): T? = wrapped.tryNext()
+    public fun tryNext(): T?
 
     /** @return the ServerCursor if available */
     public val serverCursor: ServerCursor?
-        get() = wrapped.serverCursor
 
     /** @return the ServerAddress */
     public val serverAddress: ServerAddress
+}
+
+/**
+ * The Mongo Cursor interface for change streams implementing the iterator protocol.
+ *
+ * An application should ensure that a cursor is closed in all circumstances, e.g. using a `use` statement:
+ * ```
+ *   collection.watch().cursor().use { c ->
+ *      while (c.hasNext()) {
+ *          println(c.next())
+ *      }
+ *  }
+ * ```
+ *
+ * @param T The type of documents the cursor contains
+ */
+public interface MongoChangeStreamCursor<T : Any> : MongoCursor<T> {
```

Comment:
Should this also be sealed?

Reply:
It extends a sealed interface - so implicitly is

Reply:
Have made it explicit just incase.